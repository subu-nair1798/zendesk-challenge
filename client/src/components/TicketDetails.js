import React from 'react';
import { Card } from 'semantic-ui-react';

const TicketDetail = ({ ticket }) => {
    const {
        id,
        created_at,
        subject,
        priority,
        status,
        description
    } = ticket;

    const items = [
        {
            header: 'ID',
            description: id,
            style: { width: "60px" }
        },
        {
            header: 'Subject',
            description: subject,
            style: { width: "339px" }
        },
        {
            header: 'Created at',
            description: 'Date: ' + (created_at + '').substring(0, 10) + ' | Time: ' + (created_at + '').substring(11, 16),
            style: { width: "300px" }
        },
        {
            header: 'Priority',
            description: (priority == null) ? 'Not Specified' : priority,
            style: { width: "175px" }
        },
        {
            header: 'Status',
            description: status,
            style: { width: "165px" }
        },
        {
            header: 'Description',
            description: description,
            style: { width: "1094px" }
        }
    ];

    return( <Card.Group items={items} /> );
}

export default TicketDetail;