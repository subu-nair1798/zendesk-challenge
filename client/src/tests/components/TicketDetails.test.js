import React from 'react';
import { shallow } from 'enzyme';
import { sampleResponse } from '../sampleResponse';
import TicketDetails from '../../components/TicketDetails';

describe('<TicketDetails />', () => {
    it('should render Card Group', () => {
        const wrapper = shallow(<TicketDetails ticket={sampleResponse.tickets[0]} />);
        expect(wrapper).toMatchSnapshot();
    });
});