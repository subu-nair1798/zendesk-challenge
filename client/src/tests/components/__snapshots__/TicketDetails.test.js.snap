// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketDetails /> should render Card Group 1`] = `
<CardGroup
  items={
    Array [
      Object {
        "description": 1,
        "header": "ID",
        "style": Object {
          "width": "60px",
        },
      },
      Object {
        "description": "Sample ticket: Meet the ticket",
        "header": "Subject",
        "style": Object {
          "width": "339px",
        },
      },
      Object {
        "description": "Date: 2020-11-06 | Time: 17:31",
        "header": "Created at",
        "style": Object {
          "width": "300px",
        },
      },
      Object {
        "description": "normal",
        "header": "Priority",
        "style": Object {
          "width": "175px",
        },
      },
      Object {
        "description": "open",
        "header": "Status",
        "style": Object {
          "width": "165px",
        },
      },
      Object {
        "description": "Hi Subramaniam,

This is your first ticket. Ta-da! Any customer request sent to your supported channels (email, chat, voicemail, web form, and tweet) will become a Support ticket, just like this one. Respond to this ticket by typing a message above and clicking Submit. You can also see how an email becomes a ticket by emailing your new account, support@subkaz.zendesk.com. Your ticket will appear in ticket views.

That's the ticket on tickets. If you want to learn more, check out: 
https://support.zendesk.com/hc/en-us/articles/203691476
",
        "header": "Description",
        "style": Object {
          "width": "1094px",
        },
      },
    ]
  }
/>
`;
